src/test/java/org/example/validator/ApplicationValidatorTest.java

```diff
@@ -0,0 +1,125 @@
+
+public class ApplicationValidatorTest {
+
+    AuthService authService = Mockito.mock(AuthService.class);
+    JobRoleService jobRoleService = Mockito.mock(JobRoleService.class);
+
+    ApplicationValidator applicationValidator =
+            new ApplicationValidator(authService, jobRoleService);
+
+
+    @Test
+    public void validateApplication_shouldNotThrowError_whenValidApplication() throws
+            SQLException, DoesNotExistException {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
+        );
+
+        doNothing().when(authService).checkEmailExists(applicationRequest.getEmail());
+        doNothing().when(jobRoleService).checkJobRoleIsValid(applicationRequest.getRoleId());
+
+
+        assertDoesNotThrow(() -> applicationValidator.validateApplication(applicationRequest));
+
+    }
+
+    @Test
+    public void validateApplication_shouldThrowDoesNotExistException_AuthServiceThrowsDoesNotExistException()
+    throws SQLException, DoesNotExistException {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
+        );
+
+        doThrow(DoesNotExistException.class).when(authService).checkEmailExists(applicationRequest.getEmail());
+        doNothing().when(jobRoleService).checkJobRoleIsValid(applicationRequest.getRoleId());
+
+        assertThrows(DoesNotExistException.class, () ->
+                applicationValidator.validateApplication(applicationRequest));
+    }
+
+    @Test
+    public void validateApplication_shouldThrowSQLException_AuthServiceThrowsSQLException()
+            throws SQLException, DoesNotExistException {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
+        );
+
+        doThrow(SQLException.class).when(authService).checkEmailExists(applicationRequest.getEmail());
+        doNothing().when(jobRoleService).checkJobRoleIsValid(applicationRequest.getRoleId());
+
+        assertThrows(SQLException.class, () ->
+                applicationValidator.validateApplication(applicationRequest));
+    }
+
+
+    @Test
+    public void validateApplication_shouldThrowDoesNotExistException_JobRoleServiceThrowsInvalidException()
+            throws SQLException, DoesNotExistException {
+        ApplicationRequest applicationRequest = new ApplicationRequest(
+                "user@kainos.com",
+                1,
+                "https://academy-job-portal-cvs.s3.eu-west-1.amazonaws.com/the_extra_semicolon/TestDocument.docx"
```

Comment:
View above.

Reply:
This has been corrected, I do not want to be reviewing Polar Bear's Pull Requests. They have no thumbs and don't use exception handling.